### ISSUE TITLE
[create vulnerability->general -> cwe] Click +, the list page turns without data

### ISSUE THREAD

**Issue by x-zim** [DependencyTrack/dependency-track#3856]
### Current Behavior

[create vulnerability->general -> cwe] Click +, and then click the next page inside, there is no data

### Steps to Reproduce

1.[create vulnerability->general -> cwe] Click +, then click Turn page

### Expected Behavior

Can turn the page normally and return the data normally

### Dependency-Track Version

4.11.0

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by x-zim** [comment]
@nscuro 

Issue Description
I encountered a pagination logic issue in the project. The relevant code is as follows:


```java
int pos = 0;
final var cwes = new ArrayList<Cwe>();
for (final Map.Entry<Integer, String> dictEntry : CweDictionary.DICTIONARY.entrySet()) {
    if (pagination.getOffset() > pos) {
        continue;
    }
    if (pagination.getLimit() <= pos) {
        break;
    }
    final var cwe = new Cwe();
    cwe.setCweId(dictEntry.getKey());
    cwe.setName(dictEntry.getValue());
    cwes.add(cwe);
    pos++;
}
```


Cause of the Issue
Initialization and Increment of pos:

The variable pos is initialized to 0, which causes an issue when pagination starts (i.e., when pos < pagination.getOffset()), as pos does not increment properly. This leads to the condition pagination.getOffset() being unmet for the initial entries, resulting in incorrect skipping.
Order of continue and break Statements:

The continue statement skips the current loop iteration without incrementing pos, which can cause some entries to be incorrectly skipped.
The break statement may prematurely terminate the loop before items are added to cwes, particularly when pagination.getLimit() is smaller than expected, leading to fewer elements being returned than intended.
Proposed Solution
To resolve this issue, I suggest the following simplified and more effective solution:

```java
final var cwes = new ArrayList<Cwe>();
int offset = pagination.getOffset();
int limit = pagination.getLimit();
int pos = 0;
int count = 0;

for (final Map.Entry<Integer, String> dictEntry : CweDictionary.DICTIONARY.entrySet()) {
    if (pos >= offset && count < limit) {
        final var cwe = new Cwe();
        cwe.setCweId(dictEntry.getKey());
        cwe.setName(dictEntry.getValue());
        cwes.add(cwe);
        count++;
    }
    pos++;
    if (count >= limit) {
        break;
    }
}
```

Advantages
Clearer Logic: The introduction of the count variable simplifies the pagination logic, making the code more readable.
Precise Control: The simultaneous check of offset and limit ensures that unnecessary iterations are avoided.
Improved Maintainability: This new approach aligns with common pagination patterns, making the code easier to maintain and extend.

**Comment by adamrconnah** [comment]
I am also getting this error.

I have tested on Firefox, Chrome and Edge, and all have the problem.

Steps: Vulnerabilities > Create vulnerability->General tab -> CWE > Click +. 

5 results on Page 1 are listed. The page shows "1 to 5 of 1426 rows" and 286 pages. When I click next page, no results are shown and I get an error in the Console "Uncaught TypeError: o[i.options.dataField] is undefined"

I have tried the API request via Postman too. And it returns an empty array
This is the API request: cwe?searchText=&pageSize=100&pageNumber=2

The offending parameter seems to be "pageNumber=2". If send the request cwe?searchText=&pageSize=100&pageNumber=1, it returns values. But anything other than "pageNumber=1" and it will return an empty array.


--

is there a workaround or version of DT that does not have this issue?
